src/v/cluster/partition_leaders_table.cc

```diff
@@ -80,6 +80,15 @@ partition_leaders_table::get_leader(const model::ntp& ntp) const {
 void partition_leaders_table::update_partition_leader(
   const model::ntp& ntp,
   model::term_id term,
+  std::optional<model::node_id> leader_id) {
+    // we set revision_id to invalid, this way we will skip revision check
+    update_partition_leader(ntp, model::revision_id{}, term, leader_id);
```

Comment:
why skip revision check here? it seems like in all cases the check should be something like:

```
if revisions are different: update
else if term is newer: update
```

Reply:
EDIT: oh I see I think this is an interface used in a specific scenario where revision isn't available for some reason?